File: admin/src/main/java/com/beat/admin/promotion/application/result/AdminPromotionPresignedUrlResults.java
package com.beat.admin.promotion.application.result;

import java.util.Map;

import com.beat.contracts.storage.CarouselPresignedUpload;

public final class AdminPromotionPresignedUrlResults {

	private AdminPromotionPresignedUrlResults() {
	}

	public record CarouselPresignedUrlsResult(Map<String, CarouselPresignedUpload> carouselPresignedUploads) {
		public CarouselPresignedUrlsResult {
			carouselPresignedUploads = Map.copyOf(carouselPresignedUploads);
		}
	}

	public record BannerPresignedUrlResult(String bannerPresignedUrl) {
	}
}

File: admin/src/main/kotlin/com/beat/admin/promotion/application/result/AdminPromotionPresignedUrlResults.kt
package com.beat.admin.promotion.application.result

import com.beat.contracts.storage.CarouselPresignedUpload

object AdminPromotionPresignedUrlResults {

    @JvmRecord
    data class CarouselPresignedUrlsResult(
        val carouselPresignedUploads: Map<String, CarouselPresignedUpload>,
    )

    @JvmRecord
    data class BannerPresignedUrlResult(
        val bannerPresignedUrl: String,
        val bannerImageKey: String,
    )
}